src/scope-hoisting/mangler.js

```diff
@@ -0,0 +1,69 @@
+const rename = require('./renamer');
+const t = require('babel-types');
+
+const CHARSET = (
+  'abcdefghijklmnopqrstuvwxyz' + 'ABCDEFGHIJKLMNOPQRSTUVWXYZ$_'
+).split('');
+
+/**
+ * This is a very specialized mangler designer to mangle only names in the top-level scope.
```

Comment:
Just curious, why does the mangler fail for topLevel scope? Babel minify supports toplevel mangling and I am interested to know what was the reason to have similar mangling step again here? or is it done for uglify since you are using it as the default minifier?

Reply:
The point of this mangler is to ONLY mangle the top level scope once all files are concatenation. All other scopes are mangled on a per file level.